chore: pyright

danking opened the pull request:
Closes #2668

danking committed:
chore: pyright

Signed-off-by: Daniel King <dan@spiraldb.com>

danking committed:
wip

Signed-off-by: Daniel King <dan@spiraldb.com>

danking committed:
fixes

Signed-off-by: Daniel King <dan@spiraldb.com>

danking committed:
more fixes

Signed-off-by: Daniel King <dan@spiraldb.com>

danking committed:
tests are only allowed to "import vortex as vx"

Signed-off-by: Daniel King <dan@spiraldb.com>

danking committed:
lints

Signed-off-by: Daniel King <dan@spiraldb.com>

danking committed:
include pcodec in dev deps

Signed-off-by: Daniel King <dan@spiraldb.com>

danking committed:
more fixes

Signed-off-by: Daniel King <dan@spiraldb.com>

robert3005 commented on .github/workflows/ci.yml:
```diff
@@ -56,6 +56,8 @@ jobs:
         run: uv run ruff format --check .
       - name: Python Lint - Ruff
         run: uv run ruff check .
+      - name: Python Lint - Ruff
```
this is not a ruff

robert3005 commented on pyproject.toml:
```diff
@@ -28,7 +28,9 @@ dev-dependencies = [
     "pyright>=1.1.385",
     "pytest-benchmark>=4.0.0",
     "pytest>=7.4.0",
+    "pyright>=1.1.403",
```
I think you want https://github.com/detachhead/basedpyright according to the internet

danking replied:
fml:
```
1650 errors, 32307 warnings, 0 notes
```

danking committed:
s/Ruff/PyRight/

Signed-off-by: Daniel King <dan@spiraldb.com>

danking committed:
fuckin' based

Signed-off-by: Daniel King <dan@spiraldb.com>

danking commented:
@robert3005 based

danking committed:
fix

Signed-off-by: Daniel King <dan@spiraldb.com>

danking committed:
missing scalarptype options

Signed-off-by: Daniel King <dan@spiraldb.com>

robert3005 reviewed (approved):
I will fix up the rest of the failures

robert3005 committed:
fixes

Signed-off-by: Robert Kruszewski <github@robertk.io>

robert3005 committed:
fix

Signed-off-by: Robert Kruszewski <github@robertk.io>

robert3005 committed:
fixes

Signed-off-by: Robert Kruszewski <github@robertk.io>

danking committed:
fix pyright

Signed-off-by: Daniel King <dan@spiraldb.com>

danking committed:
fix: extract Python type_aliases into importable py file

Definitions in pyi files are not imported when the corresponding modules is imported.

Signed-off-by: Daniel King <dan@spiraldb.com>

danking committed:
use future annotations and also fix IntoExpr import (which has moved)

Signed-off-by: Daniel King <dan@spiraldb.com>

danking committed:
ruff fix

Signed-off-by: Daniel King <dan@spiraldb.com>

danking committed:
wip

Signed-off-by: Daniel King <dan@spiraldb.com>

danking committed:
finally passes basedpyright

Signed-off-by: Daniel King <dan@spiraldb.com>

danking committed:
ruff

Signed-off-by: Daniel King <dan@spiraldb.com>

danking committed:
finally get it working

Signed-off-by: Daniel King <dan@spiraldb.com>

danking committed:
fix

Signed-off-by: Daniel King <dan@spiraldb.com>
